src/create-transformer.ts

```diff
@@ -3,7 +3,14 @@ import { invariant, addHiddenProp } from "./utils"
 
 export type ITransformer<A, B> = (object: A) => B
 
+export interface ITransformerParams<A, B> {
+    onCleanup?: (resultObject: B | undefined, sourceObject?: A) => void,
+    debugNameGenerator?: (sourceObject?: A) => string,
+}
+
 let memoizationId = 0
+let onCleanup: Function = undefined
```

Comment:
@ItamarShDev  Note that these variables should be declared inside `createTransformer`! Not in the module scope, otherwise all transformers will share the same cleanup and name generation code (the last call winning). Will merge anyway and fix afterwards to no longer delay the merge.